src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  function adder(a) {
+    if (a !== undefined) {
+      sum += a;
+
+      return adder;
+    } else {
+      const result = sum;
+
+      sum = 0;
```

Comment:
Resetting `sum` to 0 is correct, but make sure to consider if there are any edge cases where this could lead to unexpected behavior. For example, if `adder` is called concurrently in an asynchronous context, this could lead to race conditions.